devicetypes/plaidsystems/spruce-valve.src/spruce-valve.groovy

```diff
@@ -0,0 +1,76 @@
+/**
+ *  Copyright Plaid Systems 2020
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *
+
+11-2020
+ * valveDuration slider capability added back to presentation
+ * tabs and trim whitespace
+
+**/
+
+metadata {
+	definition (name: "Spruce Valve", namespace: "plaidsystems", author: "Plaid Systems", mnmn: "SmartThingsCommunity", vid: "558e73d3-c800-3669-b276-1d4352eda12b"){
+		capability "Actuator"
+		capability "Valve"
+		capability "Sensor"
+		capability "Health Check"
+		capability "heartreturn55003.valveDuration"
+
+		command "open"
+		command "close"
+		command "setValveDuration"
+
+		attribute "valveDuration", "NUMBER"
+	}
+}
+
+def installed() {
+	initialize()
+}
+
+def updated() {
+	initialize()
+}
+
+private initialize() {
+	//log.trace "Executing 'initialize'"
+
+	sendEvent(name: "valve", value: "closed")
+	sendEvent(name: "valveDuration", value: 5, unit: "mins")
+
+	sendEvent(name: "DeviceWatch-DeviceStatus", value: "online")
+	sendEvent(name: "healthStatus", value: "online")
+	sendEvent(name: "DeviceWatch-Enroll", value: [protocol: "cloud", scheme:"untracked"].encodeAsJson(), displayed: false)
+}
+
+def parse(String onOff) {
```

Comment:
Is it ever called?

Reply:
I need to ask again, what's the idea behind this? No command is coming directly to child device, so I don't know why `parse()` is declared here. Can you elaborate?
Is it just because you're using `child.sendEvent(name: "valve", value: onoff)` in your main DTH? If so, you don't need any "passthrough" in child DTH, this event will be registered for it right away.